app/controllers/projects/attachments_controller.rb

```diff
@@ -44,6 +46,36 @@ def create
       end
     end
 
+    def new_destroy
+      authorize! @project, to: :destroy_attachment?
+      render turbo_stream: turbo_stream.update('attachment_modal',
+                                               partial: 'delete_attachment_modal',
+                                               locals: {
+                                                 open: true
+                                               }), status: :ok
+    end
+
+    def destroy # rubocop:disable Metrics/MethodLength
+      @destroyed_attachments = ::Attachments::DestroyService.new(@project.namespace, @attachment, current_user).execute
+
+      respond_to do |format|
+        if @destroyed_attachments.count.positive?
+          status = destroy_status(@attachment, @destroyed_attachments.length)
+          format.turbo_stream do
```

Comment:
So overall this PR is great and working as expected. The message needs to be updated. If you upload a pair of files and then delete them the text in the modal reads:

![image](https://github.com/user-attachments/assets/c7c227f5-d5ad-49fe-a8cf-86921553e8cb)

The message makes sense when you are deleting a single file but here 2 files will be removed

Reply:
Fixed in 51bd2bb70e420b759680a25480266a9f73522d89 / 37e394d5d3e09568617c102666274dd1c9c089c7

Added filename(s) to the modal as well
![image](https://github.com/user-attachments/assets/b3918c2e-55f7-4da7-8715-c778af93f322)

Reply:
Perfect thanks!